### PR TITLE
LPS-206062 Check if transaction was already completed before we call commit/rollback

### PR DESCRIPTION
**Pull request by holatuwol**
https://liferay.atlassian.net/browse/LPS-206062

## Testing notes

Because this started with an investigation into `testPostSystemObjectEntryWithInvalidNestedCustomObjectEntries`, I ran the appropriate integration test to confirm that the test now passed:

```bash
../gradlew -a :apps:object:object-rest-test:testIntegration --tests SystemObjectRelatedObjectEntriesTest
```

I also followed the steps in LPS-206062 to confirm that an exception was no longer printed to the logs and tested LPS-199370 to confirm that these changes did not introduce a regression related to that ticket.

### CODE REVIEW

**Comment by holatuwol** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]
Thansk @holatuwol! Looks good to me. @ccorreagg please have a look and let's forward if possible.

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/TransactionContainerRequestFilter.java [hunk 1]
```diff
@@ -138,6 +138,10 @@ private static class TransactionCleanUpMessageObserver
 		extends AbstractPhaseInterceptor implements MessageObserver {
 
 		public void commit() {
+			if (_transactionStatusAdapter.isCompleted()) {
```
Comment (ccorreagg) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/TransactionContainerRequestFilter.java:141]: hey @holatuwol ,

Just a question, which is the difference between using the boolean private attribute `_complete` and `_transactionStatusAdapter.isCompleted()`? Could it be simplified by using just one?

I am trying to reproduce the error but by now, I can't with the log steps

Comment (holatuwol) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/TransactionContainerRequestFilter.java:141]: Yes, we could just use `_transactionStatusAdapter.isCompleted()`, I'll push an update.

Comment (holatuwol) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/TransactionContainerRequestFilter.java:141]: @ccorreagg Pushed the requested change to the branch.

**Review by ccorreagg — APPROVED** [review]

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/TransactionContainerRequestFilter.java (hunk 1 above)
Comment (ccorreagg) [reply modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/container/request/filter/TransactionContainerRequestFilter.java:141]: Thank you @holatuwol

**Comment by ccorreagg** [comment]
ci:forward

**Pull request closed without merging**